### PR TITLE
Improve safe mode messages in-game

### PR DESCRIPTION
**Pull request by GNosii**
#### Description: 

Improves the locked in Safe Mode message that is sent on PlayerJoinEvent by making it more user-friendly. This implements two different messages depending on the operator status of the player.

image
Non operator player

image
Operator player
____
#### New Nodes/Commands/ConfigOptions: 

____
#### Relevant Towny Issue ticket:

____
- [x] I have tested this pull request for defects on a server. 

By making this pull request, I represent that I have the right to waive copyright and related rights to my contribution, and agree that all copyright and related rights in my contributions are waived, and I acknowledge that the TownyAdvanced organization has the copyright to use and modify my contribution under the Towny [License](https://github.com/LlmDl/Towny/blob/master/LICENSE.md) for perpetuity.

### CODE REVIEW

**Comment by drewdorris** [comment]
Could it also send the message if the player has the "towny.admin" perm? I don't use Minecraft's OP system but do have admins with the towny.admin permission.

**Comment by GNosii** [comment]
> Could it also send the message if the player has the "towny.admin" perm? I don't use Minecraft's OP system but do have admins with the towny.admin permission.

The check for the `towny.admin` node was added and should work in theory. I'm not able to test it right now as I don't have access to an server.

File: src/com/palmergames/bukkit/towny/listeners/TownyPlayerListener.java [hunk 1]
```diff
@@ -101,11 +102,19 @@ public TownyPlayerListener(Towny instance) {
...
+		// Safe Mode Join Messages
 		if (plugin.isError()) {
-			TownyMessaging.sendMessage(player, Colors.Rose + "[Towny Error] Locked in Safe mode!");
+			// Operator or an admin.
+			if (player.isOp() || permissionSource.has(player, PermissionNodes.TOWNY_ADMIN.getNode())) {
+				TownyMessaging.sendMessage(player, Colors.Rose + "[Towny Error] Towny is locked in Safe Mode due to an error! Please check the server's console for more information.");
+				return;
+			}
+			// Regular player.
+			TownyMessaging.sendMessage(player, Colors.Rose + "[Towny Error] Towny is locked in Safe Mode due to an error! Tell an admin to check the server's console.");
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/listeners/TownyPlayerListener.java:117]:
```suggestion
			String msg = "[Towny Error] Towny is locked in Safe Mode due to an error! Tell an admin to check the server's console.";
			// Operator or an admin.
			if (TownyUniverse.getInstance().getPermissionSource().isTownyAdmin(player)) {
				msg = "[Towny Error] Towny is locked in Safe Mode due to an error! Please check the server's console for more information.";
			}
			TownyMessaging.sendErrorMsg(player, msg);
```

File: src/com/palmergames/bukkit/towny/listeners/TownyPlayerListener.java [hunk 2]
```diff
@@ -101,11 +102,19 @@ public TownyPlayerListener(Towny instance) {
 
 	@EventHandler(priority = EventPriority.NORMAL)
 	public void onPlayerJoin(PlayerJoinEvent event) {
-
+		TownyPermissionSource permissionSource = TownyUniverse.getInstance().getPermissionSource();
+
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/listeners/TownyPlayerListener.java:106]:
```suggestion

```

**Review by LlmDl — CHANGES_REQUESTED** [review]

File: src/com/palmergames/bukkit/towny/listeners/TownyPlayerListener.java (hunk 1 above)
Comment (Warriorrrr) [reply src/com/palmergames/bukkit/towny/listeners/TownyPlayerListener.java:117]: Having the error messages be translatable would also be a good idea

File: resources/english.yml [hunk 3]
```diff
@@ -1410,6 +1410,10 @@ jail_sing: 'Jail'
 msg_player_has_been_sent_to_jail_into_cell_number_x_for_x_hours_by_x: '%s has been sent to %s into cell number %s for %s hours by %s.'
 msg_resident_has_not_played_long_enough_to_be_jailed: '%s hasn''t played long enough to be jailed, wait another %s'
 
+msg_safe_mode_base: '[Towny Error] Towny is locked in Safe Mode due to an error! '
+msg_safe_mode_player: 'Tell an admin to check the server''s console.'
+msg_safe_mode_admin: 'Check the server''s console for more information.'
+
```
Comment (LlmDl) [inline resources/english.yml:1416]: Lang strings were added in twice.

File: src/com/palmergames/bukkit/towny/listeners/TownyPlayerListener.java (hunk 2 above)
Comment (LlmDl) [reply src/com/palmergames/bukkit/towny/listeners/TownyPlayerListener.java:106]: These lines are un-needed.

File: src/com/palmergames/bukkit/towny/listeners/TownyPlayerListener.java [hunk 4]
```diff
@@ -101,11 +102,21 @@ public TownyPlayerListener(Towny instance) {
...
+		// Safe Mode Join Messages
 		if (plugin.isError()) {
-			TownyMessaging.sendMessage(player, Colors.Rose + "[Towny Error] Locked in Safe mode!");
+			String tipMsg = Translation.of("msg_safe_mode_player");
+
+			// Operator or an admin.
+			if (TownyUniverse.getInstance().getPermissionSource().isTownyAdmin(player)) {
+				tipMsg = Translation.of("msg_safe_mode_admin");
+				return;
+			}
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/listeners/TownyPlayerListener.java:117]:
```suggestion
			if (TownyUniverse.getInstance().getPermissionSource().isTownyAdmin(player))
				tipMsg = Translation.of("msg_safe_mode_admin");
```

**Review by LlmDl — CHANGES_REQUESTED** [review]

File: src/com/palmergames/bukkit/towny/listeners/TownyPlayerListener.java [hunk 5]
```diff
@@ -24,6 +24,7 @@
 import com.palmergames.bukkit.towny.object.Translation;
 import com.palmergames.bukkit.towny.object.WorldCoord;
 import com.palmergames.bukkit.towny.permissions.PermissionNodes;
+import com.palmergames.bukkit.towny.permissions.TownyPermissionSource;
 import com.palmergames.bukkit.towny.permissions.TownyPerms;
```
Comment (Warriorrrr) [inline src/com/palmergames/bukkit/towny/listeners/TownyPlayerListener.java:28]: Unused import, the rest looks great.

Comment (GNosii) [reply src/com/palmergames/bukkit/towny/listeners/TownyPlayerListener.java]: removed unused import

**Review by LlmDl — APPROVED** [review]

**Pull request merged**